WalletWasabi.Gui/Tabs/TextViewModelBase.cs

```diff
@@ -0,0 +1,57 @@
+using Avalonia;
+using Avalonia.Platform;
+using ReactiveUI;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using System.Reactive.Threading.Tasks;
+using System.Text;
+using System.Threading.Tasks;
+using WalletWasabi.Gui.ViewModels;
+
+namespace WalletWasabi.Gui.Tabs
+{
+	public abstract class TextViewModelBase : WasabiDocumentTabViewModel
+	{
+		protected CompositeDisposable Disposables { get; private set; } = new CompositeDisposable();
+
+		public string _text;
+
+		public TextViewModelBase(Global global, string title, Uri target) : base(global, title)
+		{
+			Text = "";
+
+			LoadDocumentAsync(target)
+				.ToObservable()
+				.ObserveOn(RxApp.MainThreadScheduler)
+				.Subscribe(x => Text = x)
+				.DisposeWith(Disposables);
```

Comment:
@lontivero @molnard I think this Task.ToObservable is super useful. Can you take a note of this pattern?

Reply:
When you want to run an async function in the constructor, then make it ToObservable and subscribe to its result on the UI thread. Also make sure to dispose (In `OnClose` usually.)

Reply:
ACK. Why is this pattern better than simply using the old continuations with `ContinueWith`?

Reply:
Very beautiful in this particular case but not a silver bullet for every place - notice taken! 
@lontivero with `ContinueWith` wouldn't we need a `Task.Run` or an `await`?

Reply:
`ContinueWith` returns a `Task` that you can await or not. Here, for example, we are not awaiting the task returned by the `LoadDocumentAsync` method.

Reply:
Ahh forget it, `ToObservable` is implemented with a `ContinueWith` https://github.com/mono/rx/blob/74a538f6725ebc83efda4bb07d5747e8a6359e19/Rx/NET/Source/System.Reactive.Linq/Reactive/Threading/Tasks/TaskObservableExtensions.cs#L24